lib/site_prism/section.rb

```diff
@@ -15,12 +15,12 @@ def initialize(parent, root_element)
       Capybara.within(@root_element) { yield(self) } if block_given?
     end
 
-    def visible?
-      root_element.visible?
+    def page
```

Comment:
I think I've worked this out.

Is the current code calling `page.find(matcher)` and as such you've overridden page at the Section level to provide the more granular version (localised element or if that isn't found then revert back to page)

Reply:
edit: I've looked into the capybara docs and it does seem as though in the example you provided the failure is correct. But looking into the Capybara docs further it seems as though `have_text` which is an equivalent alias to `have_content` were only designed to be scoped at a page level. And infact it could potentially be fixed in the Capybara code.

Reply:
Hi, @luke-hill! Sorry for the late reply.

> only designed to be scoped at a page level

Could you please clarify, why do you think so? If this is because of the comment `Capybara::SessionMatchers`, I think this is just typo. Simple example:

*HTML*:
```
<div>
  <p class="sometext">Some text</p>
  <span>ANOTHER text</span>
</div>
```

*Code*:
```
node = page.find(".sometext")
expect(node).to have_text("ANOTHER")
```

This spec fails with the following error:
```
RSpec::Expectations::ExpectationNotMetError: expected to find text "ANOTHER" in "Some text"
```

The methods in https://github.com/teamcapybara/capybara/blob/master/lib/capybara/rspec/matchers.rb file are just matchers for rspec. 

`Capybara::DSL` module "delegates" `DSL_METHODS` to the `page` method (as you can see here: https://github.com/teamcapybara/capybara/blob/master/lib/capybara/dsl.rb#L49). I think, in Capybara terminology, the `page` method is a root level element. Therefore, I have defined `page` method in the Section class and this method returns `root_element` for this section. Now all capybara methods delegated to the section's root element. If you think that it is not clear, we can remove `Capybara::DSL` module and just "delegate" `NODE_METHODS` to the root_element (https://github.com/teamcapybara/capybara/blob/dfaf958ea7b9f02f31986f253ee10e6314853389/lib/capybara/session.rb#L41). DSL_METHODS includes NODE_METHODS, SESSION_METHODS, and MODAL_METHODS. And I think only NODE_METHODS make sense for our case. What do you think about this?

Thanks!

Reply:
I posted an update below. And in capybara. They believe this "should" already happen.

I tested this a while ago on a version of capybara much older (I originally had this issue a year or two ago), Thomas is convinced this shouldn't happen and that all methods should be auto-scoped to whatever context you are calling them on.

Could you please post an SSCCE to the issue I mentioned where the current Test Case fails (i.e. it isn't being scoped correctly)